include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+    uint32_t maxViewports;
+    uint32_t maxViewportDims[2];
+    //float                 viewportBoundsRange[2];
+    //uint32_t              viewportSubPixelBits;
+    //size_t                minMemoryMapAlignment;
+    uint32_t bufferViewAlignment;
+    uint32_t UBOAlignment;
+    uint32_t SSBOAlignment;
+    //int32_t               minTexelOffset;
+    //uint32_t              maxTexelOffset;
+    //int32_t               minTexelGatherOffset;
+    //uint32_t              maxTexelGatherOffset;
+    //float                 minInterpolationOffset;
+    //float                 maxInterpolationOffset;
+    //uint32_t              subPixelInterpolationOffsetBits;
+    //uint32_t              maxFramebufferWidth;
+    //uint32_t              maxFramebufferHeight;
+    //uint32_t              maxFramebufferLayers;
+    //VkSampleCountFlags    framebufferColorSampleCounts;
+    //VkSampleCountFlags    framebufferDepthSampleCounts;
+    //VkSampleCountFlags    framebufferStencilSampleCounts;
+    //VkSampleCountFlags    framebufferNoAttachmentsSampleCounts;
+    //uint32_t              maxColorAttachments;
+    //VkSampleCountFlags    sampledImageColorSampleCounts;
+    //VkSampleCountFlags    sampledImageIntegerSampleCounts;
+    //VkSampleCountFlags    sampledImageDepthSampleCounts;
+    //VkSampleCountFlags    sampledImageStencilSampleCounts;
+    //VkSampleCountFlags    storageImageSampleCounts;
+    //uint32_t              maxSampleMaskWords;
+    //VkBool32              timestampComputeAndGraphics;
+    float    timestampPeriodInNanoSeconds; // timestampPeriod is the number of nanoseconds required for a timestamp query to be incremented by 1 (a float because vulkan reports), use core::rational in the future
+    //uint32_t              maxClipDistances;
+    //uint32_t              maxCullDistances;
+    //uint32_t              maxCombinedClipAndCullDistances;
+    //uint32_t              discreteQueuePriorities;
+    float pointSizeRange[2];
+    float lineWidthRange[2];
+    //float                 pointSizeGranularity;
+    //float                 lineWidthGranularity;
+    //VkBool32              strictLines;
+    //VkBool32              standardSampleLocations;
+    //VkDeviceSize          optimalBufferCopyOffsetAlignment;
+    //VkDeviceSize          optimalBufferCopyRowPitchAlignment;
+    uint64_t nonCoherentAtomSize;
+            
+    /* Vulkan 1.1 Core  */
+    uint32_t subgroupSize;
+    core::bitflag<asset::IShader::E_SHADER_STAGE> subgroupOpsShaderStages;
+    bool shaderSubgroupBasic = false;
+    bool shaderSubgroupVote = false;
+    bool shaderSubgroupArithmetic = false;
+    bool shaderSubgroupBallot = false;
+    bool shaderSubgroupShuffle = false;
+    bool shaderSubgroupShuffleRelative = false;
+    bool shaderSubgroupClustered = false;
+    bool shaderSubgroupQuad = false;
+    bool shaderSubgroupQuadAllStages = false; //quadOperationsInAllStages;
+    //VkPointClippingBehavior    pointClippingBehavior;
```

Comment:
I don't think we want this:
> VK_POINT_CLIPPING_BEHAVIOR_ALL_CLIP_PLANES specifies that the primitive is discarded if the vertex lies outside any clip plane, including the planes bounding the view volume.
VK_POINT_CLIPPING_BEHAVIOR_USER_CLIP_PLANES_ONLY specifies that the primitive is discarded only if the vertex lies outside any user clip plane.

Reply:
https://github.com/KhronosGroup/OpenGL-API/issues/51
https://www.khronos.org/opengl/wiki/Vertex_Post-Processing#Clipping

So basically which one it will be depends on GL vs GLES vs Nvidia.

Reply:
Please verify:
If it's `NVIDIA || IsGLES`: `VK_POINT_CLIPPING_BEHAVIOR_USER_CLIP_PLANES_ONLY`
otherwise: `VK_POINT_CLIPPING_BEHAVIOR_ALL_CLIP_PLANES `

Reply:
looks like it yes, leave the two links in the `OpenGL` impl in the comments as justification